### PR TITLE
admin: Add circuit breaker settings to /clusters?format=json

### LINKED ISSUE #10897: admin: circuit breaker thresholds not in `/clusters` JSON
The plaintext version of `/clusters` includes the runtime interpolated value for circuit breakers, see [AdminImpl::addCircuitSettings](https://github.com/envoyproxy/envoy/blob/25ccc5f2442300bc31a4975ddff3d572fbc37519/source/server/http/admin.cc#L336-L347).

In practice:
```
$ curl -s 0:9999/clusters | grep priority::max | head -n4
my_cluster::default_priority::max_connections::1024
my_cluster::default_priority::max_pending_requests::1024
my_cluster::default_priority::max_requests::3000
my_cluster::default_priority::max_retries::3
```

The JSON version does not, see [admin/v4alpha/clusters.proto](https://github.com/envoyproxy/envoy/blob/master/api/envoy/admin/v4alpha/clusters.proto).

Unless this information is available in another admin API, and even if it is, it seems like adding it to the JSON API would be a good idea.

### PR DESCRIPTION
**Pull request by dio**
Commit Message: This adds circuit breaker settings to /clusters?format=json cluster status.

Risk Level: Low
Testing: Unit
Docs Changes: Updated
Release Notes: Added
Fixes #10897

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -76,6 +76,7 @@ New Features
 * access log: added support for :ref:`%DOWNSTREAM_PEER_FINGERPRINT_1% <config_access_log_format_response_flags>` as a response flag.
 * access log: added support for nested objects in :ref:`JSON logging mode <config_access_log_format_dictionaries>`.
 * access log: added :ref:`omit_empty_values<envoy_v3_api_field_config.core.v3.SubstitutionFormatString.omit_empty_values>` option to omit unset value from formatted log.
+* admin: added :ref:`circuit breakers settings <envoy_v3_api_msg_config.cluster.v3.CircuitBreakers>` information to GET /clusters?format=json :ref:`cluster status <envoy_v3_api_msg_admin.v3.ClusterStatus>`.
```
Comment (adisuissa) [inline docs/root/version_history/current.rst:79]: nit: also to GET /clusters (text format)

**Review by asraa — COMMENTED** [review]
Thanks! Code looks good. Question:
It's unfortunate that DEFAULT priorities don't appear in json formatting. it seems like it's all default fields are rendered (ints, enums) or none for C++ json printer... 

Hopefully this is clear to the user, but maybe that can be commented on in the API?

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (dio) [reply docs/root/version_history/current.rst:79]: We have this already. As mentioned in the issue, we’re missing only for JSOn format.

**Comment by dio** [comment]
@asraa yeah. I’ll add a comment.

**Review by asraa — APPROVED** [review]
Thanks @dio! looks good to me!

**Review by htuch — DISMISSED** [review]
LGTM, thanks!

**Comment by dio** [comment]
Thanks, @asraa, @htuch, and @adisuissa for the review! I think I need another approval since there was a conflicted file.

**Review by asraa — APPROVED** [review]
No worries! LGTM

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**